west.yml

```diff
@@ -0,0 +1,38 @@
+# SOF west manifest
+manifest:
+  defaults:
+    remote: thesofproject
+
+  remotes:
+    - name: thesofproject
+      url-base: https://github.com/thesofproject
+    - name: zephyrproject
+      url-base: https://github.com/zephyrproject-rtos
+
+  projects:
+    - name: rimage
+      repo-path: rimage
+      path: sof/rimage
+      revision: 02abc5d342a3ee6965bdc933ea1439d85d0256da
+      remote: thesofproject
+
+    - name: tomlc99
+      repo-path: tomlc99
+      path: sof/rimage/tomlc99
+      revision: e3a03f5ec7d8d33be705c5ce8a632d998ce9b4d1
+      remote: thesofproject
+
+    - name: zephyr
+      repo-path: zephyr
+      revision: 63a497280f22aea39793dd9de9731469a83ce533
+      remote: zephyrproject
+      import:
+        name-whitelist:
+          - hal_xtensa
+          - mbedtls
+          - mipi-sys-t
+          - lz4
+          - tinycrypt
+
+  self:
+    path: sof
```

Comment:
It can't rename itself, but you can after this PR gets merged clone SOF project with
`west init -m https://github.com/thesofproject/sof.git`

Reply:
You can try with(after this PR merges):
```west init -m https://github.com/thesofproject/sof.git  sof_zephyr```

Then it will clone `sof` under `sof_zephyr` folder.

You can also specify the branch you want to clone: 
```west init -m https://github.com/thesofproject/sof.git --mr $branch sof_zephyr```

Reply:
My question is: what happens if this `self: path:` is not there.

Reply:
Short answer: It blocks users from "wrong" usage of the project.
### Performed following test:
1. created new empty _sof_test_workspace_ directory.
2. executed command: `west init -m https://github.com/aborisovich/sof.git --mr use-west-manifest sof-custom-dir-name`
(which clones content of this PR using `west init` call to custom sof project directory named _sof-custom-dir-name_)
3. Executed `west update` that failed

### Results:

![image](https://user-images.githubusercontent.com/12482515/180666329-d74ed4e5-c9a8-4584-b81c-474b865255d5.png)

1. SOF project is cloned to _sof-custom-dir-name_ directory (overriden by positional `west init` argument).
2. Command `west init` fails because `west.yml` `self: path: sof` does not match positional argument. _.west_ directory is not created in _sof_test_workspace_
3. `west update` may not be run as west failed to initailize.

### Gain

User may not use broken project-tree configuration:
_sof_test_workspace/sof-custom-dir-name
sof_test_workspace/sof/rimage
sof_test_workspace/sof/rimage/tomlc99_